src/include/mpir_objects.h

```diff
@@ -156,9 +156,7 @@ typedef enum MPII_Object_kind {
     MPIR_PROCGROUP = 0xc,       /* These are internal device objects */
     MPIR_VCONN = 0xd,
     MPIR_WORKQ_ELEM = 0xe,      /* Work queue element, currently only meaningful in CH4 */
-    MPIR_GREQ_CLASS = 0xf,
-    MPIR_XPMEM_SEG = 0x10,      /* XPMEM segment, only meaningful in CH4
-                                 * when XPMEM shmmod is enabled */
+    MPIR_GREQ_CLASS = 0xf
 } MPII_Object_kind;
```

Comment:
I am checking with Ken to figure out whether we can define more object kinds here. If we cannot, you should first fix the `MPIR_XPMEM_SEG` in a single commit, then add dynamic counter pool code.